pkg/controller/applicationlayer/applicationlayer_controller.go

```diff
@@ -440,32 +461,75 @@ func getApplicationLayer(ctx context.Context, cli client.Client) (*operatorv1.Ap
 	return instance, nil
 }
 
-func (r *ReconcileApplicationLayer) isLogsCollectionEnabled(l7Spec *operatorv1.LogCollectionSpec) bool {
+func (r *ReconcileApplicationLayer) isLogsCollectionEnabled(applicationLayerSpec *operatorv1.ApplicationLayerSpec) bool {
+	l7Spec := applicationLayerSpec.LogCollection
 	return l7Spec != nil && l7Spec.CollectLogs != nil && *l7Spec.CollectLogs == operatorv1.L7LogCollectionEnabled
 }
 
+func (r *ReconcileApplicationLayer) isALPEnabled(applicationLayerSpec *operatorv1.ApplicationLayerSpec) bool {
+	return applicationLayerSpec.ApplicationLayerPolicyStatus != nil &&
+		*applicationLayerSpec.ApplicationLayerPolicyStatus == operatorv1.ApplicationLayerPolicyEnabled
+}
+
 func (r *ReconcileApplicationLayer) isWAFEnabled(applicationLayerSpec *operatorv1.ApplicationLayerSpec) bool {
 	return applicationLayerSpec.WebApplicationFirewall != nil &&
 		*applicationLayerSpec.WebApplicationFirewall == operatorv1.WAFEnabled
 }
 
-// patchFelixTproxyMode takes all application layer specs as arguments and patches felix config.
+func (r *ReconcileApplicationLayer) polsyncPathPfx(fcSpec *crdv1.FelixConfigurationSpec, al *operatorv1.ApplicationLayer) (_ string) {
```

Comment:
nit, but please avoid such abbreviations. We can afford the extra characters to make this code more readable.

`policysyncPathPrefix`

Also, could you switch these functions so that they don't use named returns? I don't think it helps the readability of these functions and isn't consistent with how we've written the rest of this code.

Reply:
@electricjesus I don't think you've addressed this comment fully ^^.